components/micro-gateway-core/src/main/ballerina/src/gateway/filters/validation_filter.bal

```diff
@@ -118,15 +124,22 @@
          if (method is string) {
             reqMethod = method;
          }
-         string resPayload = response.getJsonPayload().toString();
+         var payload = response.getJsonPayload();
+         if (payload is map<json>)  {
+              resPayload = payload.toJsonString();
+         }
          string servName = context.getServiceName();
-         printDebug(KEY_VALIDATION_FILTER, "The Response payload : " + resPayload);
-         var valResult = responseValidate(resPath, requestMethod, responseCode, resPayload, servName);
+         var valResult = responseValidate(resPath, reqMethod, responseCode, resPayload, servName);
          if (valResult is handle && stringutils:equalsIgnoreCase(valResult.toString(), VALIDATION_STATUS)) {
              return true;
          } else {
+             json newPayload = { fault: {
+                                        code: 500,
+                                        message: "Bad Response",
+                                        description: valResult.toString()
+                               } };
              response.statusCode = 500;
```

Comment:
Lets try to use the constants if available in b7a http module

Reply:
fixed.